### PR TITLE
Tweak studio docs

### PR DESCRIPTION
**Pull request by stefanvanburen**
Minor things:

* fixed a few small typos
* switched to straight quotes for consistency
* removed gif TODO (image is already a gif!)
* fixed anchor link

### CODE REVIEW

File: docs/bsr/studio.md [hunk 1]
```diff
@@ -21,24 +21,23 @@ you can:
...
-  access.
+- Create shareable links for requests defined on Buf Studio to team members
+  with access.
 
 ## Composing Requests
 
-Start by selecting the Protobuf method you’re looking to make a request with.
-With the “Select Method” menu you can choose a BSR module and use Buf Studio's
+Start by selecting the Protobuf method you're looking to make a request with.
+With the "Select Method" menu you can choose a BSR module and use Buf Studio's
```
Comment (renahlee) [inline docs/bsr/studio.md:30]: Hopefully moving forward, these should be handled by prettier, if not already. Thanks for fixing this up ❤️

File: docs/bsr/studio.md [hunk 2]
```diff
@@ -21,24 +21,23 @@ you can:
...
 fuzzy search to select the desired service and method for you request:
 
-<!-- TODO: make this into a gif -->
 <Image alt="Studio method select demo" src="/img/bsr/studio-method-select.gif" width={90} />
```
Comment (renahlee) [inline docs/bsr/studio.md:33]: This looks to be a file with JSX that I missed -- if possible, could you change the extension of this file to `.mdx` and update reference sites? (No worries if you don't want to go down that rabbit hole, just let me know :) )

**Review by renahlee — APPROVED** [review]

File: docs/bsr/studio.md (hunk 2 above)
Comment (stefanvanburen) [reply docs/bsr/studio.md:33]: D'oh, I did this but did it on main (and pushed, oops). Just merged instead.

In terms of reference sites, I did a quick `rg /studio` and didn't find any links to this page, so I think we're set.

I actually noticed that the Mermaid diagrams don't seem to work in .mdx:

image

so I'm going to merge this as-is (to fix my accidental push to main). If you know of a way to fix the Mermaid diagrams, we can definitely move over to mdx!

Comment (renahlee) [reply docs/bsr/studio.md:33]: Ah! Thank you so much for catching that. I opened a placeholder ticket for now (FE-525) :)

**Pull request merged**